tsdb/cmd/tsdb/main.go

```diff
@@ -617,9 +633,34 @@ func analyzeBlock(b tsdb.BlockReader, limit int) error {
 	return nil
 }
 
-func dumpSamples(db *tsdb.DBReadOnly, mint, maxt int64) (err error) {
+type dumpConfiguration struct {
```

Comment:
what's the convention for placing typedefs and functions? I would expects typedefs at the top of the file like in C, not sure what's expected in Go

Reply:
Yeah its a bit different to C in that usually the convention is to keep types close to where they are used. That way its easy for any maintainer to find the types. Now, some core types are typically grouped at the top of a file, but for me `dumpConfiguration` is not so much as a core file as `Configuration` would be. You find a similar pattern with the `writeBenchmark` struct on L159.